aten/src/ATen/native/mps/operations/LinearAlgebra.mm

```diff
@@ -496,26 +496,25 @@ static void linalg_inv_ex_out_mps_impl(const Tensor& A, bool check_errors, const
   using namespace mps;
   TORCH_CHECK(result.is_mps(), "Output tensor is not MPS");
   TORCH_CHECK(!A.is_complex(), "linalg_inv: not supported for complex types yet!");
-  using CachedGraph = MPSUnaryCachedGraph;
 
-  MPSStream* stream = getCurrentMPSStream();
   info.zero_();
-
   if (A.numel() == 0) {
     return;
   }
 
-  if (!result.is_contiguous()) {
-    result.unsafeGetTensorImpl()->empty_tensor_restride(MemoryFormat::Contiguous);
-  }
   auto A_sizes = A.sizes();
   int ndim = A.dim();
 
-  Tensor LU = empty_like(A);
-  Tensor identity = zeros_like(A);
+  Tensor LU = empty_like(A, MemoryFormat::Contiguous);
+  Tensor identity = zeros_like(A, MemoryFormat::Contiguous);
   Tensor pivots = empty({A_sizes.begin(), A_sizes.end() - 1}, A.options().dtype(kInt));
   (ndim == 2 ? identity.diagonal() : identity.diagonal(0, -2, -1)).fill_(1);
```

Comment:
Isn't the left statement of this ternary a noop? Convert to if statement?

Reply:
Are you sure, doesn't `identity.diagonal(0, -2, -1)).fill_(1);` have side effects?